dataplicity/client.py

```diff
@@ -95,31 +96,48 @@ def disk_poll(self):
 
         if now >= self.next_disk_poll_time:
             self.next_disk_poll_time = now + self.disk_poll_rate_seconds
-            disk_space = disk_usage('/')
+            disk_space = disk_usage("/")
 
             with self.remote.batch() as batch:
                 batch.call_with_id(
-                    'authenticate_result',
-                    'device.check_auth',
-                    device_class='tuxtunnel',
+                    "authenticate_result",
+                    "device.check_auth",
+                    device_class="tuxtunnel",
                     serial=self.serial,
-                    auth_token=self.auth_token
+                    auth_token=self.auth_token,
                 )
                 batch.call_with_id(
-                    'set_disk_space_result',
-                    'device.set_disk_space',
+                    "set_disk_space_result",
+                    "device.set_disk_space",
                     disk_capacity=disk_space.total,
-                    disk_used=disk_space.used
+                    disk_used=disk_space.used,
                 )
 
+    def tag_poll(self):
+        with self.remote.batch() as batch:
+            batch.call_with_id(
+                "authenticate_result",
+                "device.check_auth",
+                device_class="tuxtunnel",
+                serial=self.serial,
+                auth_token=self.auth_token,
+            )
+            batch.call_with_id(
+                "set_machine_defined_tags_result",
+                "device.set_machine_defined_tags",
+                tag_list=get_tag_list(),
+            )
+
     def poll(self):
         """Called at regular intervals."""
         t = time.time()
-        log.debug('poll t=%.02fs', t)
+        log.debug("poll t=%.02fs", t)
         try:
             self.disk_poll()
         except Exception as e:
             log.error("disk poll failed %s", e)
+
+        self.tag_poll()
```

Comment:
This call should be wrapped in an exception handler. It may throw exceptions if there are connectivity issues.